core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -620,12 +617,67 @@ class CivilizationInfo {
+                CivFlags.ShouldResetDiplomaticVotes.name -> {
+                    gameInfo.diplomaticVictoryVotesCast.clear()
+                    removeFlag(CivFlags.ShouldResetDiplomaticVotes.name)
+                    removeFlag(CivFlags.ShowDiplomaticVotingResults.name)
+                }
+                CivFlags.ShowDiplomaticVotingResults.name -> {
+                    addFlag(CivFlags.ShouldResetDiplomaticVotes.name, 1)
+                    
+                    if (gameInfo.civilizations.any { it.victoryManager.hasWon() } )
+                        // We have either already done this calculation, or it doesn't matter anymore, 
+                        // so don't waste resources doing it
+                        continue
+                    
+                    val results = victoryManager.calculateDiplomaticVotingResults(gameInfo.diplomaticVictoryVotesCast)
+                    val bestCiv = results.maxByOrNull { it.value }
+                    if (bestCiv == null) continue
+                    
+                    val civCount = gameInfo.civilizations.count { !it.isDefeated() }
+                    
+                    // CvGame.cpp::DoUpdateDiploVictory() in the source code of the original
+                    val votesNeededToWin = (
+                        if (civCount > 28) 0.35 * civCount
+                        else (67 - 1.1 * civCount) / 100 * civCount
+                        ).toInt()
+                    
+                    if (bestCiv.value < votesNeededToWin) continue
+                    
+                    gameInfo.civilizations.first { it.civName == bestCiv.key }.victoryManager.hasWonDiplomaticVictory = true
+                }
+            }
         }
     }
 
     fun addFlag(flag: String, count: Int) { flagsCountdown[flag] = count }
+    fun removeFlag(flag: String) { flagsCountdown.remove(flag) }
 
+    fun getTurnsBetweenDiplomaticVotings() = (15 * gameInfo.gameParameters.gameSpeed.modifier).toInt() // Dunno the exact calculation, hidden in Lua files
+    
+    fun getTurnsTillNextDiplomaticVote() = 
+        if (CivFlags.TurnsTillNextDiplomaticVote.name !in flagsCountdown) null 
+        else flagsCountdown[CivFlags.TurnsTillNextDiplomaticVote.name]
+    
+    fun mayVoteForDiplomaticVictory() =
+        CivFlags.TurnsTillNextDiplomaticVote.name in flagsCountdown.keys 
+        && flagsCountdown[CivFlags.TurnsTillNextDiplomaticVote.name] == 0 
+        && civName !in gameInfo.diplomaticVictoryVotesCast.keys
+    
+    fun diplomaticVoteForCiv(chosenCivName: String?) {
+        if (chosenCivName != null) gameInfo.diplomaticVictoryVotesCast[civName] = chosenCivName
+        addFlag(CivFlags.TurnsTillNextDiplomaticVote.name, getTurnsBetweenDiplomaticVotings())
+    }
+    
+    fun shouldShowDiplomaticVotingResults() =
+         CivFlags.ShowDiplomaticVotingResults.name in flagsCountdown.keys
+         && flagsCountdown[CivFlags.ShowDiplomaticVotingResults.name] == 0
```

Comment:
I think we can have just this check and not the "in keys" check, since if it isn't in the flags then it will just be null